backend/de.metas.payment.esr/src/main/java/de/metas/payment/esr/api/impl/ESRImportBL.java

```diff
@@ -1294,6 +1330,46 @@ public void linkBankStatementLinesByPaymentIds(@NonNull final Map<PaymentId, Ban
 
 			linkESRImportLineToBankStatement(esrImportLine, bankStatementLineRefId);
 		}
+
+		final ImmutableSet<ESRImportId> esrImportIds = extractESRImportIds(esrImportLines);
+		updateESRImportReconciledStatus(esrImportIds);
+	}
+
+	private static ImmutableSet<ESRImportId> extractESRImportIds(@NonNull final List<I_ESR_ImportLine> esrImportLines)
+	{
+		return esrImportLines.stream()
+				.map(esrImportLine -> ESRImportId.ofRepoId(esrImportLine.getESR_Import_ID()))
+				.collect(ImmutableSet.toImmutableSet());
+	}
+
+	@VisibleForTesting
+	void updateESRImportReconciledStatus(@NonNull final Set<ESRImportId> esrImportIds)
+	{
+		if (esrImportIds.isEmpty())
+		{
+			// shall NOT happen
+			return;
+		}
+
+		final ImmutableSet<ESRImportId> notReconciledESRImportIds = Services.get(IQueryBL.class)
```

Comment:
i think it's cleaner if you move this method into esrImportDAO..

Reply:
Moved